### ISSUE TITLE
[4.x]: Error with Conditions on Dropdown fields with Optgroup

### ISSUE THREAD

**Issue by tommysvr** [craftcms/cms#14224]
### Description
When editing an Entry Type with an existing condition on a Dropdown field with an Optgroup, I'm getting an `Undefined array key "value"` error.

```
[web.ERROR] [yii\base\ErrorException:2] yii\base\ErrorException: Undefined array key "value" in /var/www/html/vendor/craftcms/cms/src/fields/conditions/OptionsFieldConditionRule.php:28
```

Trying to create a new condition based on a Dropdown field with an Optgroup doesn't error, but you're not able to add any rules.

It looks to be related to this change: https://github.com/craftcms/cms/commit/6ceb9246fc42ed9e1d65b1393fa3d6a53165e333

### Steps to reproduce

1. Create a Dropdown field with an Optgroup
2. Add that to an Entry Type
3. Add a Condition to another field based on the Dropdown
4. Unable to add rules

### Expected behavior
Able to add rules like this:

![screenshot_2024_01_27 _13 29](https://github.com/craftcms/cms/assets/867566/7c6c04e4-955c-43dd-aeec-78c53c6ff106)

### Actual behavior
Unable add any rules:

![screenshot_2024_01_27 _13 33](https://github.com/craftcms/cms/assets/867566/bfdd46c8-5951-434b-aaac-7761a2638d60)

### Craft CMS version

Tested on 4.6.1 and 4.7.0

**Comment by i-just** [comment]
Thanks for all the info, Tommy! I raised a PR for this.

**Comment by brandonkelly** [comment]
Fixed in 4.7.1!
